include/ddc/chunk.hpp

```diff
@@ -131,11 +143,11 @@ class Chunk<ElementType, DiscreteDomain<DDims...>, Allocator>
             return *this;
         }
         if (this->m_internal_mdspan.data_handle()) {
-            std::allocator_traits<
-                    Allocator>::deallocate(m_allocator, this->data_handle(), this->size());
+            m_allocator.deallocate(this->data_handle(), this->size());
         }
         static_cast<base_type&>(*this) = std::move(static_cast<base_type&>(other));
         m_allocator = std::move(other.m_allocator);
+        m_label = std::move(other.m_label);
```

Comment:
can we not retrieve the label already stored in the allocation?

Reply:
I cannot find such a public function, all I know is here: https://kokkos.github.io/kokkos-core-wiki/API/core/c_style_memory_management/malloc.html?highlight=kokkos_malloc

Reply:
It might not be stored at all if we switch to allocate, maybe that's the best option.